backend/controllers/cs_customer_sync.go

```diff
@@ -0,0 +1,144 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/ARO-HCP/internal/database"
+	"github.com/Azure/ARO-HCP/internal/ocm"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+type Controller interface {
+	Run(ctx context.Context, threadiness int) error
+}
+
+type hcpClusterKey struct {
+	subscriptionID  string
+	resourceGroupID string
+	hcpClusterID    string
+}
+
+type clusterServiceToCustomerSync struct {
+	cosmosClient database.DBClient
+
+	clusterServiceClient ocm.ClusterServiceClientSpec
+
+	// queue is where incoming work is placed to de-dup and to allow "easy"
+	// rate limited requeues on errors
+	queue workqueue.TypedRateLimitingInterface[hcpClusterKey]
+}
+
+func NewClusterServiceToCustomerSyncController(cosmosClient database.DBClient, clusterServiceClient ocm.ClusterServiceClientSpec) Controller {
+	c := &clusterServiceToCustomerSync{
+		cosmosClient:         cosmosClient,
+		clusterServiceClient: clusterServiceClient,
+		queue: workqueue.NewTypedRateLimitingQueueWithConfig(
```

Comment:
For which events do you want to be triggered? CosmosDB has changefeeds, unclear if useful to you.